pyworkflow/template.py

```diff
@@ -108,13 +111,36 @@ def _replaceFields(self):
         for field in self.params.values():
             self.content[field.getIndex()] = field.getValue()
 
+    def getParams(self):
+        return self.params
+
+    def setParamValue(self, alias, newValue):
```

Comment:
Just for my curiosity, I've tried to execute a template filling all the parameters, but repeating one of them. The resulting behavior was that the form appeared instead of running the template. I think that this method should check that the number of parameters introduced isn't higher than the number of different parameter alias contained in the template file.